test/functional/lua/vim_spec.lua

```diff
@@ -319,9 +320,39 @@ describe('lua stdlib', function()
 
       return b.x[1] == 1 and b.x[2] == 2 and b.y == 5 and count == 2
              and tostring(a) ~= tostring(b)
-    ]])
+    ]]))
+
+    ok(exec_lua([[
+      local a = {}
+      local b = vim.deepcopy(a)
+
+      local count = 0
```

Comment:
We don't need to do it now but later we could add `vim.tbl_count()` or something.